go-controller/pkg/metrics/metrics.go

```diff
@@ -166,6 +166,12 @@ func checkPodRunsOnGivenNode(clientset kubernetes.Interface, label, k8sNodeName
 	return false, fmt.Errorf("the Pod matching the label %q doesn't exist on this node %s", label, k8sNodeName)
 }
 
+var ovnRegistry = prometheus.NewRegistry()
```

Comment:
Can this be a parameter of `RegisterOvnMetrics` ?
I think it would be more deterministic than having a function changing the value of a global variable. The caller would have a way to drive the registry used, but in a more clear way and without side effects